[JIT] fix resolving of functions in torch/functional. fix compilation of torch.stft

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #33818 [JIT] Add support for list()
* #33783 [JIT] add support for torch.norm
* #33737 [JIT] add support for torch.cdist
* #33736 [JIT] add support for lu_unpack
* #33724 [JIT] add support for torch.lu to torchscript
* **#33504 [JIT] fix resolving of functions in torch/functional. fix compilation of torch.stft**

Fix resolution fo functions that are bound onto torch in torch/functional.py. This does not fix compilation of all of those functions, those will be done in follow ups. Does torch.stft as a start.

Fixes #21478

Differential Revision: [D20014591](https://our.internmc.facebook.com/intern/diff/D20014591)

Commit:
fix resolving of functions in torch/functional

[ghstack-poisoned]

Commit:
Update on "[JIT] fix resolving of functions in torch/functional"




[ghstack-poisoned]

Commit:
Update on "[JIT] fix resolving of functions in torch/functional. fix compilation of torch.stft"


Fix resolution fo functions that are bound onto torch in torch/functional.py. This does not fix compilation of all of those functions, those will be done in follow ups. Does torch.stft as a start.

Fix for https://github.com/pytorch/pytorch/issues/21478

[ghstack-poisoned]

Commit:
Update on "[JIT] fix resolving of functions in torch/functional. fix compilation of torch.stft"


Fix resolution fo functions that are bound onto torch in torch/functional.py. This does not fix compilation of all of those functions, those will be done in follow ups. Does torch.stft as a start.

Fix for https://github.com/pytorch/pytorch/issues/21478

[ghstack-poisoned]

driazati commented on torch/functional.py:
```diff
@@ -290,6 +290,7 @@ def meshgrid(*tensors, **kwargs):
     return torch._C._VariableFunctions.meshgrid(tensors)
 
 
+_VF_stft = torch._C._VariableFunctions.stft
```
Can you make this use [`_VF.py`](https://github.com/pytorch/pytorch/blob/1969898647bf33c0b1fd2625c02075d2d543f9e2/torch/nn/_VF.py#L6)? You'll probably have to move that file somewhere else but it will make it so we can just do the below instead of imports for every function

https://github.com/pytorch/pytorch/blob/cfb4862673303a2725d815c0338435f6b6c699fb/torch/nn/functional.py#L13

driazati commented on torch/jit/_builtins.py:
```diff
@@ -83,8 +83,25 @@
     (torch.nn.init._no_grad_zero_, "aten::_no_grad_zero_"),
     (torch._C._get_tracing_state, "aten::_get_tracing_state"),
     (warnings.warn, "aten::warn"),
+    (torch._C._VariableFunctions.stft, "aten::stft")
```
if you do the above you shouldn't need this here, instances in `nn/functional.py` resolve to their `aten::` equivalent already 

https://github.com/pytorch/pytorch/blob/cfb4862673303a2725d815c0338435f6b6c699fb/torch/nn/functional.py#L805-L806

eellison replied:
For some reason I had to add _VF.stft here, now sure why. But I moved _VF so that i can just use `_VF.stft` inline.

driazati reviewed (approved):
Looks good to go after a couple mechanical changes

Commit:
Update on "[JIT] fix resolving of functions in torch/functional. fix compilation of torch.stft"


Fix resolution fo functions that are bound onto torch in torch/functional.py. This does not fix compilation of all of those functions, those will be done in follow ups. Does torch.stft as a start.

Fixes #21478

[ghstack-poisoned]

Commit:
Update on "[JIT] fix resolving of functions in torch/functional. fix compilation of torch.stft"


Fix resolution fo functions that are bound onto torch in torch/functional.py. This does not fix compilation of all of those functions, those will be done in follow ups. Does torch.stft as a start.

Fixes #21478

Differential Revision: [D20014591](https://our.internmc.facebook.com/intern/diff/D20014591)

[ghstack-poisoned]

Commit:
Update on "[JIT] fix resolving of functions in torch/functional. fix compilation of torch.stft"


Fix resolution fo functions that are bound onto torch in torch/functional.py. This does not fix compilation of all of those functions, those will be done in follow ups. Does torch.stft as a start.

Fixes #21478

Differential Revision: [D20014591](https://our.internmc.facebook.com/intern/diff/D20014591)

[ghstack-poisoned]

Commit:
Update on "[JIT] fix resolving of functions in torch/functional. fix compilation of torch.stft"


Fix resolution fo functions that are bound onto torch in torch/functional.py. This does not fix compilation of all of those functions, those will be done in follow ups. Does torch.stft as a start.

Fixes #21478

Differential Revision: [D20014591](https://our.internmc.facebook.com/intern/diff/D20014591)

[ghstack-poisoned]

Commit:
Update on "[JIT] fix resolving of functions in torch/functional. fix compilation of torch.stft"


Fix resolution fo functions that are bound onto torch in torch/functional.py. This does not fix compilation of all of those functions, those will be done in follow ups. Does torch.stft as a start.

Fixes #21478

Differential Revision: [D20014591](https://our.internmc.facebook.com/intern/diff/D20014591)

[ghstack-poisoned]

Commit:
Update on "[JIT] fix resolving of functions in torch/functional. fix compilation of torch.stft"


Fix resolution fo functions that are bound onto torch in torch/functional.py. This does not fix compilation of all of those functions, those will be done in follow ups. Does torch.stft as a start.

Fixes #21478

Differential Revision: [D20014591](https://our.internmc.facebook.com/intern/diff/D20014591)

[ghstack-poisoned]

Commit:
Update on "[JIT] fix resolving of functions in torch/functional. fix compilation of torch.stft"


Fix resolution fo functions that are bound onto torch in torch/functional.py. This does not fix compilation of all of those functions, those will be done in follow ups. Does torch.stft as a start.

Fixes #21478

Differential Revision: [D20014591](https://our.internmc.facebook.com/intern/diff/D20014591)

[ghstack-poisoned]

Commit:
Update on "[JIT] fix resolving of functions in torch/functional. fix compilation of torch.stft"


Fix resolution fo functions that are bound onto torch in torch/functional.py. This does not fix compilation of all of those functions, those will be done in follow ups. Does torch.stft as a start.

Fixes #21478

Differential Revision: [D20014591](https://our.internmc.facebook.com/intern/diff/D20014591)

[ghstack-poisoned]
